### ISSUE TITLE
Remove experiment specific references from Base Experiments

### ISSUE THREAD

**Issue by ngupta23** [pycaret/pycaret#2282]
**Is your feature request related to a problem? Please describe.**
pycaret_experiment.py, supervised_experiment.py other base experiment classes have switched for task-specific items. This makes the code harder to understand. e.g. 

```python
if self._ml_usecase == MLUsecase.TIME_SERIES:
```

**Describe the solution you'd like**
Create a child class from "supervised experiment" - one for `time_series` and one for all other tasks. In this child class, add methods for doing these and call them from base classes. This would be a cleaner alternative.

**Describe alternatives you've considered**
Leave as is, but this makes it harder for new folks to be onboarded.

**Comment by ngupta23** [comment]
This one is related to the work that @Yard1 did in https://github.com/pycaret/pycaret/pull/3170. Once this is merged, we can implement this as well.

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
